sycl/test-e2e/KernelCompiler/kernel_compiler.cpp

```diff
@@ -0,0 +1,165 @@
+//==- kernel_compiler.cpp --- kernel_compiler extension tests  -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: ocloc
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// the new spec for the kernel_compiler opens the door to supporting several
+// different source languages. But, initially, OpenCL Kernels are the only ones
+// supported. This test is limited to that (thus the cm-compiler requirement)
+// but in the future it may need to broken out into other tests.
+
+#include <sycl/sycl.hpp>
+
+auto constexpr CLSource = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0);
+  out[i] = in[i]*2 + 100;
+}
+__kernel void her_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0);
+  out[i] = in[i]*5 + 1000;
+}
+)===";
+
+auto constexpr BadCLSource = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0) +  no semi-colon!!
+  out[i] = in[i]*2 + 100;
+}
+)===";
+/*
+Compile Log:
+1:3:34: error: use of undeclared identifier 'no'
+  size_t i = get_global_id(0) +  no semi-colon!!
+                                 ^
+1:3:36: error: expected ';' at end of declaration
+  size_t i = get_global_id(0) +  no semi-colon!!
+                                   ^
+                                   ;
+
+Build failed with error code: -11
+
+=============
+
+*/
+
+void testSyclKernel(sycl::queue &Q, sycl::kernel Kernel, int multiplier,
+                    int added) {
+  constexpr int N = 4;
+  sycl::cl_int InputArray[N] = {0, 1, 2, 3};
+  sycl::cl_int OutputArray[N] = {};
+
+  sycl::buffer<int, 1> InputBuf(InputArray, sycl::range<1>(N));
+  sycl::buffer<int, 1> OutputBuf(OutputArray, sycl::range<1>(N));
```

Comment:
```suggestion
  sycl::buffer InputBuf(InputArray, sycl::range<1>(N));
  sycl::buffer OutputBuf(OutputArray, sycl::range<1>(N));
```

The `int` here should be `cl_int`.  However, you can also just remove the template parameters and let them be deduced via CTAD.

Reply:
I made these same changes in the other test file as well.